accname/name/comp_name_from_content.html

```diff
@@ -2,35 +2,183 @@
+<h3 data-expectedlabel="label" data-testname="heading name from content" class="ex">label</h3>
+<a href="#" data-expectedlabel="label" data-testname="link name from content" class="ex">label</a><br>
+<br>
+
+<h1>simple with ::before</h1>
+<button data-expectedlabel="before label" data-testname="button name from content with ::before" class="ex simple-before">label</button><br>
+<h3 data-expectedlabel="before label" data-testname="heading name from content with ::before" class="ex simple-before">label</h3>
+<a href="#" data-expectedlabel="before label" data-testname="link name from content with ::before" class="ex simple-before">label</a><br>
+<br>
+
+<h1>simple with ::after</h1>
+<button data-expectedlabel="label after" data-testname="button name from content with ::after" class="ex simple-after">label</button><br>
+<h3 data-expectedlabel="label after" data-testname="heading name from content with ::after" class="ex simple-after">label</h3>
+<a href="#" data-expectedlabel="label after" data-testname="link name from content with ::after" class="ex simple-after">label</a><br>
+<br>
+
+<h1>simple with ::before and ::after</h1>
+<button data-expectedlabel="before label after" data-testname="button name from content with ::before and ::after" class="ex simple-before simple-after">label</button><br>
+<h3 data-expectedlabel="before label after" data-testname="heading name from content with ::before and ::after" class="ex simple-before simple-after">label</h3>
+<a href="#" data-expectedlabel="before label after" data-testname="link name from content with ::before and ::after" class="ex simple-before simple-after">label</a><br>
+<br>
+
+<h1>no space joiners ::before and ::after</h1>
+<button data-expectedlabel="nospacelabelnospace" data-testname="button name from content no space joiners ::before and ::after" class="ex no-space">label</button><br>
+<h3 data-expectedlabel="nospacelabelnospace" data-testname="heading name from content no space joiners ::before and ::after" class="ex no-space">label</h3>
+<a href="#" data-expectedlabel="nospacelabelnospace" data-testname="link name from content no space joiners ::before and ::after" class="ex no-space">label</a><br>
+<br>
+
+<h1>Arabic right-to-left (RTL) with ::before and ::after</h1>
+<div dir="rtl" lang="ar">
+  <button data-expectedlabel="من قبل اسم بعد" data-testname="button name from content with ::before and ::after in rtl" class="ex simple-before simple-after">اسم</button><br>
+  <h3 data-expectedlabel="من قبل اسم بعد" data-testname="heading name from content with ::before and ::after in rtl" class="ex simple-before simple-after">اسم</h3>
+  <a href="#" data-expectedlabel="من قبل اسم بعد" data-testname="link name from content with ::before and ::after in rtl" class="ex simple-before simple-after">اسم</a><br>
+</div>
+<br>
+
+<h1><a href="https://drafts.csswg.org/css-content/#alt">Alternative Text for  CSS content (previously `alt:`)</a> in pseudo-elements</h1>
```

Comment:
@cookiecrook Somewhat related to this test is the usage of "Alternative Text" as opposed to "Text Alternative" in the [CSS Generated Content](https://drafts.csswg.org/css-content/#alt) spec. Should we always opt to use "text alternative" to better align with the WCAG/ARIA specs? If you agree, I can raise an issue here: https://github.com/w3c/csswg-drafts/issues.

Reply:
No concern; I did file an issue to potentially address the one instance of "alternative text" in the ARIA spec:https://github.com/w3c/aria/issues/2059.

I view this as less of an issue on web since the technique of supplying an `alt` attribute is often synonymous with the accessibility requirement itself (providing a text alternative) and it is broadly/easily understood. If spec alignment is desirable, I would definitely advocate for usage of "text alternative" per WCAG/ARIA.